src/cmd_line/commands/file.ts

```diff
@@ -32,109 +32,94 @@ export class FileCommand extends node.CommandBase {
   get arguments(): IFileCommandArguments {
     return this._arguments;
   }
-
-  getActiveViewColumn(): vscode.ViewColumn {
-    const active = vscode.window.activeTextEditor;
-
-    if (!active) {
-      return vscode.ViewColumn.One;
+  async execute(): Promise<void> {
+    if (this._arguments.bang) {
+      await vscode.commands.executeCommand('workbench.action.files.revert');
+      return;
     }
 
-    return active.viewColumn!;
-  }
-
-  getViewColumnToRight(): vscode.ViewColumn {
-    const active = vscode.window.activeTextEditor;
+    // Need to do this before the split since it looses the activeTextEditor
+    let editorFilePath = vscode.window.activeTextEditor!.document.uri.fsPath;
 
-    if (!active) {
-      return vscode.ViewColumn.One;
+    // Do the split if requested
+    let split = false;
+    if (this._arguments.position === FilePosition.NewWindowVerticalSplit) {
+      await vscode.commands.executeCommand('workbench.action.splitEditorRight');
+      split = true;
     }
-
-    switch (active.viewColumn) {
-      case vscode.ViewColumn.One:
-        return vscode.ViewColumn.Two;
-      case vscode.ViewColumn.Two:
-        return vscode.ViewColumn.Three;
+    if (this._arguments.position === FilePosition.NewWindowHorizontalSplit) {
+      await vscode.commands.executeCommand('workbench.action.splitEditorDown');
+      split = true;
     }
 
-    return active.viewColumn!;
-  }
-
-  async execute(): Promise<void> {
-    if (this._arguments.bang) {
-      await vscode.commands.executeCommand('workbench.action.files.revert');
-      return;
-    }
+    let hidePreviousEditor = async function() {
+      if (split === true) {
+        await vscode.commands.executeCommand('workbench.action.previousEditor');
+        await vscode.commands.executeCommand('workbench.action.closeActiveEditor');
```

Comment:
Why do we close the previous editor?

Reply:
This is to reflect the behaviour of Vim. 

In the case where we split specifying a file name (`vsplit bar.c`, `split bar.c`) or create an empty new file (`:new`, `:vnew`), we start by splitting the current editor which results in two splits with the same editor/file. We then create the empty file or load the specified file. 

So assuming the following command ran in the `code.go` editor

```
:vsplit bar.c
```

If we do not close the previous editor we end up with this

<img width="557" alt="screen shot 2018-08-27 at 15 48 30" src="https://user-images.githubusercontent.com/97381/44682186-b785d480-aa10-11e8-850b-7ef7d1337953.png">

But we would rather have

<img width="535" alt="screen shot 2018-08-27 at 15 50 02" src="https://user-images.githubusercontent.com/97381/44682255-e734dc80-aa10-11e8-99f8-8bb4f2191940.png">

Reply:
Ah, I see what you mean. Thanks for the explanation.